Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2023 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Analysis.ODE.PicardLindelof
+import Mathlib.Geometry.Manifold.InteriorBoundary
+import Mathlib.Geometry.Manifold.MFDeriv
+
+/-!
+# Integral curves of vector fields on a manifold
+
+For any continuously differentiable vector field on a manifold `M` and any chosen interior point
+`x₀ : M`, there exists an integral curve `γ : ℝ → M` such that `γ t₀ = x₀` and the tangent vector of
+`γ` at `t` coincides with the vector field at `γ t` for all `t` within an open interval around `t₀`.
+
+As a corollary, such an integral curve exists for any starting point `x₀` if `M` is a manifold
+without boundary.
+
+## Main definition
+
+- **IsIntegralCurve γ v t₀ x₀**: If `v : M → TM` is a vector field on `M` and `x : M`,
+`IsIntegralCurveAt γ v t₀ x₀` means `γ : ℝ → M` is a differentiable integral curve of `v` with
+`γ x₀ = t₀`.
+
+## Tags
+
+integral curve, vector field, local existence
+-/
+
+open scoped Manifold
+open Set
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] [CompleteSpace E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M]
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace ℝ E']
+  {H' : Type*} [TopologicalSpace H'] {I' : ModelWithCorners ℝ E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M'] [SmoothManifoldWithCorners I' M']
+  {v : (x : M) → TangentSpace I x} {x₀ : M}
+  (hv : ContMDiffAt I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)) x₀) (t₀ : ℝ)
+
+/-- If `v : M → TM` is a vector field on `M` and `x : M`, `IsIntegralCurveAt γ v t₀ x₀` means
+  `γ : ℝ → M` is a differentiable integral curve of `v` with `γ x₀ = t₀`. -/
+def IsIntegralCurveAt (γ : ℝ → M) (v : (x : M) → TangentSpace I x) (t₀ : ℝ) (x₀ : M) :=
+  γ t₀ = x₀ ∧ ∃ ε > (0 : ℝ), ∀ (t : ℝ), t ∈ Ioo (t₀ - ε) (t₀ + ε) →
+    HasMFDerivAt 𝓘(ℝ, ℝ) I γ t ((1 : ℝ →L[ℝ] ℝ).smulRight (v (γ t)))
+
+variable {t₀}
+
+lemma IsIntegralCurveAt.comp_add {γ : ℝ → M} (hγ : IsIntegralCurveAt γ v t₀ x₀) (dt : ℝ) :
```

Comment:
Probably some variant of those lemmas, yes, but I'm going to avoid the scope creep for now. The `HasMFDerivAt` API can also use some extra helpful lemmas about simple operations. For another day.